drivers/temperature/adt7420/adt7420.h

```diff
@@ -96,38 +113,73 @@
 #define ADT7420_OP_MODE_SHUTDOWN	3
 
 /* ADT7420 default ID */
+#define ADT7320_DEFAULT_ID		0xC3
 #define ADT7420_DEFAULT_ID		0xCB
 
+
 /******************************************************************************/
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
+enum dev_interface {
+	SPI,
+	I2C,
+};
+
+struct adt7420_chip_info {
+	uint8_t			resolution;
+	enum dev_interface  communication;
+};
+
+enum adt7420_type {
+	ID_ADT7410,
+	ID_ADT7420,
+	ID_ADT7422,
+	ID_ADT7310,
+	ID_ADT7311,
+	ID_ADT7312,
+	ID_ADT7320
+};
+
 struct adt7420_dev {
 	/* I2C */
 	i2c_desc	*i2c_desc;
+	/* SPI */
+	spi_desc	*spi_desc;
+	/* Device Settings */
+	enum adt7420_type active_device;
 	/* Device Settings */
 	uint8_t		resolution_setting;
+
 };
 
 struct adt7420_init_param {
 	/* I2C */
 	i2c_init_param	i2c_init;
+	/* SPI */
+	spi_init_param	spi_init;
 	/* Device Settings */
 	uint8_t		resolution_setting;
+	/* Device Settings */
+	enum adt7420_type	active_device;
 };
 
+extern const struct adt7420_chip_info chip_info[];
+
 /******************************************************************************/
 /************************ Functions Declarations ******************************/
 /******************************************************************************/
 
-/*! Reads the value of a register. */
-uint8_t adt7420_get_register_value(struct adt7420_dev *dev,
-				   uint8_t register_address);
+/*! Write to input shift register SPI interface. */
+uint16_t set_shift_reg(struct adt7420_dev *dev,
```

Comment:
I want do any modification in the current API.
When an application calls adt7420_get_register_value or adt7420_set_register_value, the driver will check the dev type and then will choose to call either adt7420_i2c_get_register_value or adt7420_spi_get_register_value

Reply:
If you really think some changes are needed to the API functions, please do it in an other commit an motivate the reason.